src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.RuntimeOnly.cs

```diff
@@ -1631,6 +1610,123 @@ public static void XmlSchemaTest()
         Assert.Equal(originalmapping.XsdTypeNamespace, newmapping.XsdTypeNamespace);
     }
 
+    public class DateAndTimeSchemaWrapper
+    {
+        public DateOnly DateValue;
+        public TimeOnly TimeValue;
+        [XmlElement(DataType = "time")] // Force xs:time via secondary primitive mapping
+        public TimeOnly TimeAsXsdTime;
+    }
+
+    [Fact]
+    public static void XmlSchema_DateOnly_TimeOnly_Primitives_And_Time_Override()
+    {
+        // Helper to find simpleType by name.
+        static XmlSchemaSimpleType GetSimple(XmlSchema schema, string name) =>
+            schema.Items
+                .OfType<XmlSchemaSimpleType>()
+                .FirstOrDefault(st => st.Name == name)
+                ?? throw new Xunit.Sdk.XunitException($"SimpleType '{name}' not found.");
+
+        // Export the schema for the 'DateAndTimeSchemaWrapper' class above
+        var schemas = new XmlSchemas();
+        var exporter = new XmlSchemaExporter(schemas);
+        var importer = new XmlReflectionImporter();
+        var mapping = importer.ImportTypeMapping(typeof(DateAndTimeSchemaWrapper));
+        exporter.ExportTypeMapping(mapping);
+
+        // Compile schemas to ensure validity.
+        schemas.Compile((_, e) => Assert.Fail("Schema compile error: " + e.Message), true);
+
+        // Ensure the Urt schema (with dateOnly/timeOnly) came along with the DateAndTimeSchemaWrapper definition
+        const string urtNs = "http://microsoft.com/wsdl/types/";
+        XmlSchema? urtSchema = schemas.Cast<XmlSchema>().FirstOrDefault(s => s.TargetNamespace == urtNs);
+        Assert.NotNull(urtSchema);
```

Comment:
Did this not work without the Cast? XmlSchemas implements the interface `IEnumerable<XmlSchema>` so I would have expected that not to be needed. It also derived from CollectionBase which transitively implements IEnumerable. I wonder if you could avoid the Cast call if you specified the generic type on FirstOrDefault? Just feels odd to have to cast.